### PR TITLE
Use spring instead of leap for integration tests

### LINKED ISSUE #276: Change tests dependency to spring
Update test dependancies for CDT to use Spring. Will warn and proceed if Spring libraries are not found. 
- change env `leap_DIR` to `spring_DIR`
- look for package `spring`
- change cmake config `EOSIO_VERSION_MIN` and `EOSIO_VERSION_SOFT_MAX`
- `README.md` and `docs/09_tutorials`

### PR DESCRIPTION
**Pull request by heifner**
## Change Description

- Update integration tests to use Spring instead of Leap

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

Resolves #276

### CODE REVIEW

**Review by spoonincode — APPROVED** [review]

File: tests/integration/multi_index_tests.cpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ using namespace eosio::testing;
 
 BOOST_AUTO_TEST_SUITE(multi_index_tests)
 
-// this test is copy from leap test_api_multi_index
+// this test is copy from spring test_api_multi_index
```
Comment (linh2931) [inline tests/integration/multi_index_tests.cpp:21]: The `s` in this `spring` should be capitalized as `Spring`.

File: tests/integration/CMakeLists.txt [hunk 2]
```diff
@@ -1,10 +1,10 @@
 cmake_minimum_required( VERSION 3.5 )
 
-set(EOSIO_VERSION_MIN "3.1")
+set(EOSIO_VERSION_MIN "1.0")
```
Comment (linh2931) [inline tests/integration/CMakeLists.txt:3]: Should it still be called `EOSIO_VERSION`, now that it goes back to 1.0. But no need to change for now.

**Review by linh2931 — APPROVED** [review]

File: tests/integration/multi_index_tests.cpp (hunk 1 above)
Comment (heifner) [reply tests/integration/multi_index_tests.cpp:21]: Why?

Comment (greg7mdp) [inline tests/integration/multi_index_tests.cpp:21]:
```suggestion
// this test is copied from spring test_api_multi_index
```

**Review by greg7mdp — APPROVED** [review]

**Pull request merged**